src/main/java/no/unit/nva/fileupload/CreateUploadHandler.java

```diff
@@ -0,0 +1,114 @@
+import nva.commons.exceptions.ApiGatewayException;
+import nva.commons.handlers.ApiGatewayHandler;
+import nva.commons.handlers.RequestInfo;
+import nva.commons.utils.Environment;
+import nva.commons.utils.JacocoGenerated;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.core.MediaType;
+import java.util.UUID;
+
+import static java.util.Objects.requireNonNull;
+import static org.apache.http.HttpStatus.SC_CREATED;
+
+public class CreateUploadHandler extends ApiGatewayHandler<CreateUploadRequestBody, CreateUploadResponseBody> {
+
+    private static final Logger logger = LoggerFactory.getLogger(CreateUploadHandler.class);
+    private final transient AmazonS3 s3Client;
+    private final transient String bucketName;
+
+    /**
+     * Default constructor for CreateUploadHandler.
+     */
+    @JacocoGenerated
+    public CreateUploadHandler() {
+        this(new Environment());
+    }
+
+    /**
+     * Constructor for CreateUploadHandler.
+     *
+     * @param environment environment reader
+     */
+    @JacocoGenerated
+    public CreateUploadHandler(Environment environment) {
+        this(
+                environment,
+                S3Utils.createAmazonS3Client(environment.readEnv(S3Constants.AWS_REGION_KEY)),
+                environment.readEnv(S3Constants.S3_UPLOAD_BUCKET_KEY)
+        );
+    }
+
+    /**
+     * Construct for lambda eventhandler to create an upload request for S3.
+     */
+    public CreateUploadHandler(Environment environment, AmazonS3 s3Client, String bucketName) {
+        super(CreateUploadRequestBody.class, environment, logger);
+        this.bucketName = bucketName;
+        this.s3Client = s3Client;
+
+    }
+
+    /**
+     * Extracting metadata from the given file resource.
+     * @param requestBody incoming parameters
+     * @return Metadata to be stored with file on S3
+     */
+    protected ObjectMetadata toObjectMetadata(CreateUploadRequestBody requestBody) {
+        ObjectMetadata objectMetadata = new ObjectMetadata();
+        objectMetadata.setContentMD5(null);
+        objectMetadata.setContentDisposition(toContentDisposition(requestBody.getFilename()));
+        objectMetadata.setContentType(requestBody.getMimetype());
+        return objectMetadata;
+    }
+
+    private String toContentDisposition(String filename) {
+        return String.format("filename=\"%s\"", filename);
+    }
+
+
+    @Override
+    protected CreateUploadResponseBody processInput(CreateUploadRequestBody input, RequestInfo requestInfo,
+                                                    Context context) throws ApiGatewayException {
+        validate(input);
+
+        String keyName = UUID.randomUUID().toString();
+        InitiateMultipartUploadRequest initRequest =
+                new InitiateMultipartUploadRequest(bucketName, keyName, toObjectMetadata(input));
+        InitiateMultipartUploadResult initResponse = s3Client.initiateMultipartUpload(initRequest);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code manually manages uploads or downloads. The SDK can automatically handle these operations. Consider using the SDK to automatically handle this instead. SDK class: TransferManager. See [Using TransferManager for Amazon S3 Operations](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/examples-s3-transfermanager.html).